src/reducers/versions.tsx

```diff
@@ -509,6 +513,47 @@ export const getRelativeDiffAnchor = ({
   return null;
 };
 
+export type GetRelativeDiffParams = {
+  _findRelativePathWithDiff?: typeof findRelativePathWithDiff;
+  _getRelativeDiffAnchor?: typeof getRelativeDiffAnchor;
+  currentAnchor: string | void;
+  diff: DiffInfo;
+  pathList: string[];
+  position?: RelativePathPosition;
+  version: Version;
+};
+
+type RelativeDiffResult = {
+  anchor: string | null;
+  path: string | null;
+};
+
+export const getRelativeDiff = ({
+  _findRelativePathWithDiff = findRelativePathWithDiff,
+  _getRelativeDiffAnchor = getRelativeDiffAnchor,
+  currentAnchor,
+  diff,
+  pathList,
+  position = RelativePathPosition.next,
+  version,
+}: GetRelativeDiffParams): RelativeDiffResult => {
```

Comment:
Instead of returning `RelativeDiffResult` can't this just return a `link: string`? I'd think you would handle an anchor or path just the same, like `history.push(link)`, no?

Reply:
I'm not sure about this. I wouldn't want to have to calculate the full path for the link in here, but I could simply return a string that includes the path with or without an anchor appended. E.g., either `file1.js` or `file1.js#D2`.

My plan, though, was to use this information to construct a new location using data in `router.location`, similar to what we do in [`viewVersionFile`](https://github.com/mozilla/addons-code-manager/blob/4bf9a43cae9cb4cc062e5f00b9a174f3bdd2ba56/src/reducers/versions.tsx#L830-L833), and therefore it seemed like it would be more handy to have the `path`, which ends up being a query parameter, and the `anchor`, which ends up being the `hash`, as separate properties.

Does that make sense?